src/main/java/no/nav/pto/veilarbportefolje/persononinfo/PdlService.java

```diff
@@ -52,18 +52,23 @@ public void hentOgLagreBrukerDataPaBarn(Fnr fnrBarn) {
 
     public void lagreBrukerData(Fnr fnrPerson, PDLPerson personData) {
         pdlPersonRepository.upsertPerson(fnrPerson, personData);
-
-        List<BarnUnder18Aar> barn = new ArrayList<>();
-        if (personData.getForeldreansvar() != null && !personData.getForeldreansvar().isEmpty()) {
-            personData.getForeldreansvar().forEach(barnFnr -> {
-                PDLPersonBarn barnPdl = pdlClient.hentBrukerBarnDataFraPdl(barnFnr);
-
-                if (barnPdl.isErIlive()) {
-                    barn.add(new BarnUnder18Aar(barnFnr, barnPdl.getFodselsdato(), barnPdl.getDiskresjonskode()));
-                }
-            });
+        try {
+            List<BarnUnder18Aar> barn = new ArrayList<>();
+            if (personData.getForeldreansvar() != null && !personData.getForeldreansvar().isEmpty()) {
+                personData.getForeldreansvar().forEach(barnFnr -> {
+                    PDLPersonBarn barnPdl = pdlClient.hentBrukerBarnDataFraPdl(barnFnr);
+
+                    secureLog.debug("Barn info: " + barnPdl.getFodselsdato() + ", " + barnPdl.isErIlive());
+                    if (barnPdl.isErIlive()) {
+                        barn.add(new BarnUnder18Aar(barnFnr, barnPdl.getFodselsdato(), barnPdl.getDiskresjonskode()));
+                    }
+                });
+            }
+            barnUnder18AarService.lagreBarnOgForeldreansvar(fnrPerson, barn);
+        }
+        catch (Exception e){
+            secureLog.warn("Can't fetch data about barn for person: " + fnrPerson, e);
```

Comment:
Maybe this can be edited to "Unable to fetch data about one or more barn for person", since it could have fetched the three first children successfully?

Reply:
Sounds good!